src/P/Monad.hs

```diff
@@ -67,3 +69,11 @@ f <$!> m = do
 {-# INLINE (<$!>) #-}
 
 #endif
+
+-- | Strict version of 'Control.Monad.liftM2'.
+liftM2' :: (Monad m) => (a -> b -> c) -> m a -> m b -> m c
+liftM2' f a b = do
+  !x <- a
+  !y <- b
+  return (f x y)
```

Comment:
Yeah so the version you have is strict in the arguments `x` and `y`, but is still lazy in the return type, so it doesn't force `f` to be evaluated.

You probably don't need the strictness in `x` and `y` because if that's what you want you you can call it via `liftM2' (\ !a !b -> f a b)`, or if the arguments are required to produce the result of `f` then just calling `liftM2' f` is enough.

Your initial version is equivalent to `liftM2 (\ !a !b -> f a b)` (i.e. the built in `liftM2`, but called with a function which is strict in both arguments)